TEST_APPS/device/nfcapp/nfctestshim.cpp

```diff
@@ -0,0 +1,314 @@
+/*
+ * Copyright (c) 2018 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <stdio.h>
+#include <stdarg.h>
+#include <cstring>
+#include <string.h>
+#include "mbed.h"
+#include "mbed-client-cli/ns_cmdline.h"
+#include "nfc/ndef/common/Text.h"
+#include "nfc/ndef/common/URI.h"
+#include "nfc/ndef/common/Mime.h"
+#include "nfc/ndef/MessageBuilder.h"
+#include "nfc/ndef/common/util.h"
+#include "nfc/nfcdefinitions.h"
+
+#include "NFCEEPROMDriver.h"
+#include "nfcprocess.h"
+#include "nfctestshim.h"
+#include "SmartPoster.h"
+
+using mbed::Span;
+using mbed::nfc::ndef::MessageBuilder;
+using mbed::nfc::ndef::RecordType;
+using mbed::nfc::ndef::common::span_from_cstr;
+using mbed::nfc::ndef::common::Text;
+using mbed::nfc::ndef::common::URI;
+using mbed::nfc::nfc_rf_protocols_bitmask_t;
+
+// statics
+char NFCTestShim::long_string[0x2000];
```

Comment:
are they meant to be static? wasn't there already a buffer this big defined somewhere?